### ISSUE TITLE
Aleatoric uncertainty losses

### ISSUE THREAD

**Issue by jimfleming** [fomorians/pyoneer#16]
This paper discusses two loss functions for regression and classification losses which capture aleatoric uncertainty: https://arxiv.org/abs/1703.04977

```
def aleatoric_regression_loss(dist, labels, weights):
    losses = (labels - dist.loc)**2 / (2 * dist.scale**2)
    losses += (1 / 2) * log(dist.scale**2)
    loss = compute_weighted_loss(losses, weights)
    return loss
```

```
def aleatoric_classification_loss(dist, labels, weights):
    # requires sampling
    return loss
```

There are lots of ways of formulating these and maybe even something more direct using distributions. Open to suggestions.
